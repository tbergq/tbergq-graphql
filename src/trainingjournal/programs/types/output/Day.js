// @flow

import { GraphQLObjectType, GraphQLString } from 'graphql';
import GlobalID from '@kiwicom/graphql-global-id';
import {
  connectionDefinitions,
  connectionArgs,
  type ConnectionArguments,
  connectionFromArray,
} from 'graphql-relay';

import ExerciseConnection from './Exercise';
import type { Day as DayType, Exercise } from '../../dataloaders/ProgramLoader';
import { nodeInterface } from '../../../../types/node/node';
import { register } from '../../../../types/node/typeStore';

export const Day = new GraphQLObjectType({
  name: 'Day',
  interfaces: [nodeInterface],
  fields: {
    id: GlobalID(({ id, _id }) => id ?? _id),
    name: {
      type: GraphQLString,
    },

    exercises: {
      type: ExerciseConnection,
      args: {
        ...connectionArgs,
      },
      resolve: ({ exercises }: DayType, args: ConnectionArguments) => {
        return exercises == null ? null : connectionFromArray<Exercise>(exercises, args);
      },
    },
  },
});

const { connectionType: DayConnection } = connectionDefinitions({
  nodeType: Day,
});

register('Day', Day);

export default DayConnection;
