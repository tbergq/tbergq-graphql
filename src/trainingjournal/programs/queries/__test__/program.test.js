// @flow

import { generateTestsFromFixtures } from '@kiwicom/test-utils';

import executeTestQuery from '../../../../common/services/executeTestQuery';
import Day from '../../datasets/day.json';

fetch.mockResponses([JSON.stringify(Day)]);

jest.mock('../../../db/ProgramModel.js', () => ({
  getProgram: () =>
    Promise.resolve({
      _id: 67,
      name: 'Getting back in 2 it',
      date: '2017-08-08T09:20:11Z',
    }),
}));

generateTestsFromFixtures(`${__dirname}/__fixtures__`, input =>
  executeTestQuery(input),
);
