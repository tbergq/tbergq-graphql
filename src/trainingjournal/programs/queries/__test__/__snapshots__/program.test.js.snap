// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: program.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
query program {
  program(programId: "UHJvZ3JhbTo2Nw==") {
    id
    name
    weeks(first: 1) {
      edges {
        node {
          name
          id
          days(first: 1) {
            edges {
              node {
                id
                name
                exercises(first: 1) {
                  edges {
                    node {
                      id
                      set
                      reps
                      breakTime
                      baseExercise {
                        name
                        id
                        videoLink
                        description
                        muscleGroup {
                          id
                          name
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "program": {
      "id": "UHJvZ3JhbTo2Nw==",
      "name": "Getting back in 2 it",
      "weeks": {
        "edges": [
          {
            "node": {
              "name": "Week 1",
              "id": "V2Vlazo1Nw==",
              "days": {
                "edges": [
                  {
                    "node": {
                      "id": "RGF5OjE0Mg==",
                      "name": "Day 1",
                      "exercises": {
                        "edges": [
                          {
                            "node": {
                              "id": "RXhlcmNpc2U6ODQ4",
                              "set": "4",
                              "reps": "5",
                              "breakTime": "3",
                              "baseExercise": {
                                "name": "Kneb√∏y",
                                "id": "QmFzZUV4ZXJjaXNlOjE=",
                                "videoLink": "https://www.youtube.com/watch?v=Dy28eq2PjcM",
                                "description": "",
                                "muscleGroup": {
                                  "id": "TXVzY2xlR3JvdXA6OQ==",
                                  "name": "Quads"
                                }
                              }
                            }
                          }
                        ]
                      }
                    }
                  }
                ]
              }
            }
          }
        ]
      }
    }
  }
}
`;
